### PR TITLE
Adds QR Code functionality to oms Reports

### LINKED ISSUE #4818: QR Codes in Reports
## Is your feature request related to a problem? Please describe 👀

Add the ability to use QR Codes in open-mSupply reports

## Describe the solution you'd like 🎁

Probably a custom function we can call from tera to generate a QR Code.

### Describe alternatives you've considered 💭

Attempted with Javascript but doesn't run in print preview

### Moneyworks Jobcode 🧰

COMSCUST

### PR DESCRIPTION
**Pull request by jmbrunskill**
Fixes #4818

# 👩🏻‍💻 What does this PR do?

Adds a tera function to create a QR code as base64 png.

## 💌 Any notes for the reviewer?

Should I switch to SVG? WOuld remove a dependancy on `image` might be a smaller output too?

See:
https://crates.io/crates/qrcode

# 🧪 Testing

- [ ] Can be tested with this reports branch https://github.com/msupply-foundation/open-msupply-reports/pull/94

# 📃 Documentation

- [ ] Should add developer docs for how to use QR Codes...

### CODE REVIEW

File: server/service/Cargo.toml [hunk 1]
```diff
@@ -56,6 +56,9 @@ telnet = "0.2.1"
 tempfile = "3.10.1"
 scraper = "0.20.0"
 umya-spreadsheet = "2.0.0"
+qrcode = "0.14"
+image = "0.25"
```
Comment (jmbrunskill) [inline server/service/Cargo.toml:60]: Switching to SVG would remove this dep

**Review by mark-prins — APPROVED** [review]
🚢  legend! nice and easy to use and very handy functionality

I've been toying with the svg implementation which is less code, fewer dependencies and maybe fewer bytes transmitted to the report. but I'm having character encoding issues.. so I'm happy to run with the binary version!

side note: the error message here is pretty wild
Screenshot 2024-09-20 at 10 13 54 AM

**Comment by mark-prins** [comment]
the svg implementation looks good too..
here's a suggestion for the [report repo](https://github.com/msupply-foundation/open-msupply-reports/compare/92-comoros-packing-list(suggestion)...92-comoros-packing-list(suggestion)(suggestion-svg))
and for [this repo](https://github.com/msupply-foundation/open-msupply/compare/4818-qr-codes-in-reports...4818-qr-codes-in-reports(suggestion-svg))
😄

File: server/service/src/report/report_service.rs [hunk 2]
```diff
@@ -408,6 +409,22 @@ fn generate_report(
         context.insert("arguments", &arguments);
     }
     let mut tera = tera::Tera::default();
+
+    tera.register_function(
+        "qr_code",
```
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply server/service/src/report/report_service.rs:414]: Added some docs in the mSupply reports repo. Not an obvious place in this repo. @fergie-nz Can we add this hen we update the standard reports docs next week?

**Pull request merged**